### PR TITLE
Add test to check that modules are invoked with the expected parameters.

### PR DESCRIPTION
**Pull request by tomchop**

### CODE REVIEW

File: data/recipes/upload_web_ts.json [hunk 1]
```diff
@@ -15,19 +15,20 @@
...
-	      "analyzers": "browser_search,browser_timeframe,account_finder,phishy_domains,evtx_gap,login,win_crash,safebrowsing,chain,"
+            "wait_for_timelines": "@wait_for_timelines",
+	        "analyzers": "browser_search,browser_timeframe,account_finder,phishy_domains,evtx_gap,login,win_crash,safebrowsing,chain,"
         }
     }, {
         "wants": ["TimesketchExporter"],
         "name": "TimesketchEnhancer",
         "args": {
-	    "wait_for_analyzers": "@wait_for_analyzers",
+	        "wait_for_analyzers": "@wait_for_analyzers",
```
Comment (giovannt0) [inline data/recipes/upload_web_ts.json:25]: Formatting looks odd here?

File: dftimewolf/lib/collectors/azure.py [hunk 2]
```diff
@@ -40,7 +40,10 @@ class AzureCollector(module.BaseModule):
   _ANALYSIS_VM_CONTAINER_ATTRIBUTE_NAME = 'Analysis VM'
   _ANALYSIS_VM_CONTAINER_ATTRIBUTE_TYPE = 'text'
 
-  def __init__(self, state: DFTimewolfState, critical: bool=False) -> None:
+  def __init__(self,
+               state: DFTimewolfState,
+               name: Optional[str]=None,
```
Comment (giovannt0) [inline dftimewolf/lib/collectors/azure.py:45]: `name` is missing from the docstring.

File: tests/cli/main_tool.py [hunk 3]
```diff
@@ -36,3 +38,28 @@ def testToolWithArbitraryRecipe(self):
...
+    for recipe in tool._recipes_manager.GetRecipes():
+      tool._state.LoadRecipe(recipe.contents, dftimewolf_recipes.MODULES)
+      for module in recipe.contents['modules']:
+        runtime_name = module.get('runtime_name', module['name'])
+        setup_func = tool.state._module_pool[runtime_name].SetUp
+        expected_args = set(inspect.getfullargspec(setup_func).args)
+        expected_args.remove('self')
+        provided_args = set(module['args'])
+
+        self.assertEqual(
```
Comment (giovannt0) [inline tests/cli/main_tool.py:62]: Do we need to be concerned by argument order?
Say, I have a recipe that takes `--arg1` and `--arg2`. Due to `--`, both are optional. If I pass `dftimewolf myrecipe --arg1=foo --arg2=bar` or `dftimewolf myrecipe --arg2=bar --arg1=foo` will it work?

**Review by giovannt0 — APPROVED** [review]
Couple nits, otherwise LGTM

File: tests/cli/main_tool.py (hunk 3 above)
Comment (tomchop) [reply tests/cli/main_tool.py:62]: They all end up being passed as kwargs, so the order doesn't really matter so long as the names match.

Now that all recipes are passing all parameters (optional or not), a good question to ask is whether we still want positional vs. keyword parameters in the module's SetUp() functions, or if we should just consistently have them be positional, and specify the default value in the recipe. (I would favor the latter)

Comment (giovannt0) [reply tests/cli/main_tool.py:62]: The latter SGTM

**Pull request merged**